migration.md

```diff
@@ -0,0 +1,301 @@
+# Migration Guide From v4 to v5
+
+## Summary
```

Comment:
Can you write a short summary of the changes? The below is already good, but it feels more like something you would read after upgrading and seeing compiler errors. It would be nice to have something that explains the changes without going into specifics. Like: v5 changes the internal representation of some types and will require migration. We provide methods to automate the migration process. Etc.  Etc. Basically, explain what is new and what has changed.

Reply:
We should also advise devs to be careful and do a lot of manual testing before shipping an update to users.